### PR TITLE
Added Meteor Reward for Hitting Player With a Starudst

### PR DESCRIPTION
**Pull request by wlong123**
## Overview

Now when a player hits another player's planet with one of their stardust, there is a 1 in 10 chance of a meteor stardust appearing on the screen. The meteor stardust is currently just colored grey (might be a good idea to add a specific texture for this). Moving a meteor stardust to a player's core results in a meteor shower for all players. A meteor shower causes 6 stardust to go on the screen, 4 of them are the correct color for the player who caused the meteor shower and the other 2 are randomly colored.

## Changes Made

- Added enum for stardust type
- Added new message for sending powerups
- Added new vector to stardust queue for storing powerups to be applied

### CODE REVIEW

File: source/CIGameScene.cpp [hunk 1]
```diff
@@ -204,6 +206,8 @@ void GameScene::update(float timestep) {
         _networkMessageManager->sendMessages();
         _gameUpdateManager->processGameUpdate(_stardustContainer, _planet, _opponent_planets, dimen);
     }
+    
+    processSpecialStardust(dimen, _stardustContainer);
```
Comment (henryz2000) [inline source/CIGameScene.cpp:210]: why do we do this twice?

File: source/CIGameScene.cpp [hunk 2]
```diff
@@ -264,3 +268,32 @@ void GameScene::addStardust(const Size bounds) {
...
+
+        switch (stardust->getStardustType()) {
+            case StardustModel::Type::METEOR:
+                CULog("METEOR SHOWER!");
+                stardustQueue->addStardust(stardust->getColor(), bounds);
+                stardustQueue->addStardust(stardust->getColor(), bounds);
+                stardustQueue->addStardust(stardust->getColor(), bounds);
+                stardustQueue->addStardust(stardust->getColor(), bounds);
+                stardustQueue->addStardust(CIColor::getRandomColor(), bounds);
+                stardustQueue->addStardust(CIColor::getRandomColor(), bounds);
```
Comment (henryz2000) [inline source/CIGameScene.cpp:291]: maybe we should add like a percent chance model to this if we ever decide to change it

File: source/CIGameScene.cpp (hunk 1 above)
Comment (wlong123) [reply source/CIGameScene.cpp:210]: whoops forgot to take that out

File: source/CIGameScene.cpp (hunk 2 above)
Comment (wlong123) [reply source/CIGameScene.cpp:291]: Do you mean like percent chance for colors or number of stardust? or both?

**Comment by wlong123** [comment]
Also what do people think we should do if a meteor goes off the screen? Right now it will just get sent as a regular stardust but should it send as a special stardust so people can send powerups to other players?

File: source/CIGameScene.cpp (hunk 2 above)
Comment (henryz2000) [reply source/CIGameScene.cpp:289]: either or works

**Review by steiner26 — APPROVED** [review]
Looks good to me, I think it would be nice if we could send the special stardust to other players but its not super urgent. We could just add the stardust type into the StardustSent message

**Pull request merged**